bin/test.sh

```diff
@@ -61,15 +64,24 @@ BOOKINFO_DEPLOYMENTS="details-v1 productpage-v1 ratings-v1 reviews-v1 reviews-v2
 if [ "$SKIP_SETUP" -ne 1 ]; then
     kubectl create ns bookinfo || true
     # We use the first namespace with sidecar injection enabled to determine the control plane's namespace.
-    ISTIO_CONTROL=$(kubectl get namespaces -o=jsonpath='{$.items[:1].metadata.labels.istio-env}' -l istio-env)
+    # Fails in KIND.
+    if [ -z "$ISTIO_CONTROL" ]; then
+        ISTIO_CONTROL=$(kubectl get namespaces -o=jsonpath='{$.items[:1].metadata.labels.istio-env}' -l istio-env)
+    fi
     ISTIO_CONTROL=${ISTIO_CONTROL:-istio-control}
 
-    kubectl -n bookinfo delete -f samples/bookinfo/platform/kube/bookinfo.yaml --ignore-not-found
-    kubectl -n bookinfo delete -f samples/bookinfo/networking/destination-rule-all.yaml --ignore-not-found
-    kubectl -n bookinfo delete -f samples/bookinfo/networking/bookinfo-gateway.yaml --ignore-not-found
+    if [ -z "$SKIP_DELETE" ]; then
+        kubectl -n bookinfo delete -f samples/bookinfo/platform/kube/bookinfo.yaml --ignore-not-found
+        kubectl -n bookinfo delete -f samples/bookinfo/networking/destination-rule-all.yaml --ignore-not-found
+        kubectl -n bookinfo delete -f samples/bookinfo/networking/bookinfo-gateway.yaml --ignore-not-found
 
-    kubectl label ns bookinfo istio-env=${ISTIO_CONTROL} --overwrite
+        kubectl label ns bookinfo istio-env=${ISTIO_CONTROL} --overwrite
+    fi
 
+    # Must skip if testing with global inject
```

Comment:
why?  Defect, or a TODO, or by design?

Reply:
There is a long discussion - it's more of a limitations on how we inject. 
You can use the label to select an environment if global inject is enabled - but you can't select
the environment that has the global on. And the fix is too complicated to be worth it.